replica/database.cc

```diff
@@ -1031,10 +1037,29 @@ bool database::update_column_family(schema_ptr new_schema) {
     return columns_changed;
 }
 
+void database::clean_disposed_column_families() {
+    dblog.info("clean_disposed_column_families");
+    for (auto it = _disposed_column_families.cbegin(); it != _disposed_column_families.cend(); /* no increment */ ) {
+        dblog.info("clean_disposed_column_families: {}", it->first );
+        if (it->second.use_count() == 1) {
+            dblog.info("clean_disposed_column_families: removing {}", it->first );
+            _disposed_column_families.erase(it++);
+        } else {
+            ++it;
+        }
+    }
+}
+
 void database::remove(const table& cf) noexcept {
     auto s = cf.schema();
     auto& ks = find_keyspace(s->ks_name());
+    auto lw_table_shared_ptr = _column_families.find(s->id());
+    _disposed_column_families[s->ks_name() + "." + s->cf_name()] = lw_table_shared_ptr->second;
     _column_families.erase(s->id());
+    clean_disposed_column_families();
+//    Just FYI - below commented out code causes scylla to stall
+//    while (lw_table_shared_ptr.use_count() > 0)
+//        sleep(1);
```

Comment:
Of course this will stall, it is not yielding execution control back to the reactor, which cannot schedule other tasks. The current task will consume the entire task-quota dedicated to the current scheduling groups slice, hence causing a stall.